api/src/main/java/com/github/retrooper/packetevents/protocol/packettype/PacketType.java

```diff
@@ -598,145 +651,145 @@ public static class Play {
              */
-            CHAT_COMMAND,
+            CHAT_COMMAND(WrapperPlayClientChatCommand.class),
 
             /**
              * Added with 1.19.1
              */
-            CHAT_ACK,
+            CHAT_ACK(WrapperPlayClientChatAck.class),
 
             /**
              * Added with 1.19.3
              */
-            CHAT_SESSION_UPDATE,
+            CHAT_SESSION_UPDATE(WrapperPlayClientChatSessionUpdate.class),
 
             /**
              * Added with 1.20.2
              */
-            CHUNK_BATCH_ACK,
+            CHUNK_BATCH_ACK(WrapperPlayClientChunkBatchAck.class),
             /**
              * Added with 1.20.2
              */
-            CONFIGURATION_ACK,
+            CONFIGURATION_ACK(WrapperPlayClientConfigurationAck.class),
             /**
              * Added with 1.20.2
              */
-            DEBUG_PING,
+            DEBUG_PING(WrapperPlayClientDebugPing.class),
 
             /**
              * Added with 1.20.3
              */
-            SLOT_STATE_CHANGE,
+            SLOT_STATE_CHANGE(WrapperPlayClientSlotStateChange.class),
 
             /**
              * Added with 1.20.5
              */
-            CHAT_COMMAND_UNSIGNED,
+            CHAT_COMMAND_UNSIGNED(WrapperPlayClientChatCommandUnsigned.class),
             /**
              * Added with 1.20.5
              */
-            COOKIE_RESPONSE,
+            COOKIE_RESPONSE(WrapperPlayClientCookieResponse.class),
             /**
              * Added with 1.20.5
              */
-            DEBUG_SAMPLE_SUBSCRIPTION,
+            DEBUG_SAMPLE_SUBSCRIPTION(WrapperPlayClientDebugSampleSubscription.class),
 
             /**
              * Added with 1.21.2
              */
-            CLIENT_TICK_END,
+            CLIENT_TICK_END(WrapperPlayClientClientTickEnd.class),
             /**
              * Added with 1.21.2
              */
-            SELECT_BUNDLE_ITEM,
+            SELECT_BUNDLE_ITEM(WrapperPlayClientSelectBundleItem.class),
             /**
              * Added with 1.21.2, based on {@link #STEER_VEHICLE}
              */
-            PLAYER_INPUT,
+            PLAYER_INPUT(WrapperPlayClientPlayerInput.class),
 
             /**
              * Added with 1.21.4
              */
-            PICK_ITEM_FROM_BLOCK,
+            PICK_ITEM_FROM_BLOCK(WrapperPlayClientPickItemFromBlock.class),
             /**
              * Added with 1.21.4
              */
-            PICK_ITEM_FROM_ENTITY,
+            PICK_ITEM_FROM_ENTITY(WrapperPlayClientPickItemFromEntity.class),
             /**
              * Added with 1.21.4
              */
-            PLAYER_LOADED,
+            PLAYER_LOADED(WrapperPlayClientPlayerLoaded.class),
             ;
 
             private static int INDEX = 0;
             private static final Map<Byte, Map<Integer, PacketTypeCommon>> PACKET_TYPE_ID_MAP = new HashMap<>();
             private final int[] ids;
 
-            Client() {
+            Client(@Nullable Class<?> clazz) {
```

Comment:
```suggestion
            Client(@Nullable Class<? extends PacketWrapper<?>> clazz) {
```

Also, this doesn't seem to be saved + exposed as a field